### PR TITLE
Improve error messages from slivers.

### PR DESCRIPTION
**Pull request by Hixie**
Also, fix some logic in SliverPadding.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/sliver_padding.dart [hunk 1]
```diff
@@ -162,16 +162,14 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
       to: mainAxisPadding + childLayoutGeometry.scrollExtent,
     );
     final double mainAxisPaddingPaintExtent = beforePaddingPaintExtent + afterPaddingPaintExtent;
+    final double paintExtent = math.min(
+      beforePaddingPaintExtent + math.max(childLayoutGeometry.paintExtent, childLayoutGeometry.layoutExtent + afterPaddingPaintExtent),
+      constraints.remainingPaintExtent,
+    );
```
Comment (abarth) [inline packages/flutter/lib/src/rendering/sliver_padding.dart]: Is this part of the change testable?

**Comment by abarth** [comment]
LGTM w/ test if the padding change is observable.

File: packages/flutter/lib/src/rendering/sliver_padding.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/rendering/sliver_padding.dart]: Previously, the layoutExtent would occasionally (due to floating point math) end up slightly bigger than the paintExtent. That prevents that from happening. I don't know how to actually cause it to happen. It was intermittently happening on that flickering dropdown test you saw me look at.

I'm happy to add a test if you have any ideas for how to actually reproduce the problem reliably.

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/8378

**Comment by Hixie** [comment]
landing on red because redness was a transient error on the windows bot

**Pull request merged**